frontend/packages/knative-plugin/src/components/add/EventSource.tsx

```diff
@@ -73,10 +80,14 @@ const EventSource: React.FC<Props> = ({
     data: {
       [typeEventSource.toLowerCase()]: getEventSourceData(typeEventSource.toLowerCase()),
     },
+    yamlData: !_.isEmpty(eventYaml) ? safeDump(eventYaml) : '',
   };
 
   const createResources = (rawFormData: any): Promise<K8sResourceKind> => {
-    const knEventSourceResource = getEventSourceResource(rawFormData);
+    const knEventSourceResource = getEventSourcesDepResource(
+      rawFormData,
+      !isKnownEventSourceSelected(rawFormData.type),
```

Comment:
Why do you need to do this check here? Why can't this util function do this itself?

Reply:
updated util function to handle the check